Code/GraphMol/MolStandardize/Charge.cpp

```diff
@@ -377,24 +381,36 @@ ROMol *Uncharger::uncharge(const ROMol &mol) {
       }
     }
   }
-  // Neutralize positive charges
-  std::vector<unsigned int> p_idx_matches;
-  for (const auto &match : p_matches) {
-    for (const auto &pair : match) {
-      p_idx_matches.push_back(pair.second);
-    }
+
+  // Neutralize cations until there is no longer a net charge remaining:
+  int netCharge = 0;
+  for (const auto &at : omol->atoms()) {
+    netCharge += at->getFormalCharge();
   }
-  for (const auto &idx : p_idx_matches) {
-    Atom *atom = omol->getAtomWithIdx(idx);
-    if (!atom->getNumExplicitHs()) {
-      // atoms from places like Mol blocks are normally missing explicit Hs:
-      atom->setNumExplicitHs(atom->getTotalNumHs());
+
+  if (netCharge > 0) {
+    // Neutralize positive charges
```

Comment:
Probably obvious, but perhaps:

```
Neutralize positive charges where hydrogens can be adjusted
```

Reply:
it's easy to be a bit more explicit.
Fixed